### PR TITLE
[ci] Add more details when showing node info

### PR DESCRIPTION
**Pull request by driazati**
This adds some more information to help debug when there are infra problems with Jenkins, notably:
* More Jenkins environment variables: https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#using-environment-variables
* EC2 metadata
* System level information (disk space, CPUs)

### CODE REVIEW

**Review by jroesch — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -93,10 +93,8 @@ def per_exec_ws(folder) {
 def init_git() {
   // Add more info about job node
   sh (
-    script: """
-     echo "INFO: NODE_NAME=${NODE_NAME} EXECUTOR_NUMBER=${EXECUTOR_NUMBER}"
-     """,
-     label: 'Show executor node info',
+    script: './task/scripts/task_show_node_info.sh',
```
Comment (Mousius) [inline Jenkinsfile:96]:
```suggestion
    script: './tests/scripts/task_show_node_info.sh',
```

**Review by Mousius — CHANGES_REQUESTED** [review]
Found a fairly dangerous typo but otherwise this only adds information.

File: Jenkinsfile (hunk 1 above)
Comment (Mousius) [reply Jenkinsfile:96]: (Side note: makes total sense to expect this not to live under `tests`, it'd be good start moving stuff to a root level `scripts`/`tools`/`stuff` folder given the variety of scripts we're adding)

File: tests/scripts/task_show_node_info.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+echo "$BUILD_NUMBER"
+echo "$WORKSPACE"
+
+echo "===== EC2 INFO ====="
+# See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+curl -w '\n' -fsSL http://169.254.169.254/latest/meta-data/ami-id || echo failed
+curl -w '\n' -fsSL http://169.254.169.254/latest/meta-data/instance-id || echo failed
+curl -w '\n' -fsSL http://169.254.169.254/latest/meta-data/instance-type || echo failed
+curl -w '\n' -fsSL http://169.254.169.254/latest/meta-data/hostname || echo failed
+curl -w '\n' -fsSL http://169.254.169.254/latest/meta-data/public-hostname || echo failed
```
Comment (Mousius) [inline tests/scripts/task_show_node_info.sh:34]: One clean up you could do here is just wrap it in a function `get_metadata` or similar but it's a minor thing.

File: tests/scripts/task_show_node_info.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+echo "===== RUNNER INFO ====="
+df --human-readable
+lscpu
```
Comment (Mousius) [inline tests/scripts/task_show_node_info.sh:38]: Might be worth adding `free` ?

**Review by Mousius — APPROVED** [review]
@driazati kicked off a branch test to check this all runs otherwise LGTM!

**Pull request merged**